core/src/main/java/bio/terra/pearl/core/service/workflow/RegistrationService.java

```diff
@@ -140,13 +177,26 @@ protected PreregistrationResponse validatePreRegResponseId(UUID preRegResponseId
         return preRegResponse;
     }
 
+    public String generateGUID(String userName, EnvironmentName environmentName) {
```

Comment:
this method could be renamed -- it's not really generating a GUID
```suggestion
    public String generateGovernedUsername(String proxyUserName, EnvironmentName environmentName) {
```